introspection-engine/connectors/sql-introspection-connector/src/misc_helpers.rs

```diff
@@ -204,32 +203,31 @@ pub(crate) fn calculate_backrelation_field(
         on_delete,
     });
 
-    let other_is_unique = || {
-        let table = schema.table_bang(&model.name);
-
-        match &relation_field.database_names.len() {
-            0 => table.is_column_unique(&relation_field.name),
-            1 => {
-                let column_name = relation_field.database_names.first().unwrap();
-                table.is_column_unique(column_name)
-            }
-            _ => table
-                .indices
-                .iter()
-                .any(|i| i.columns == relation_field.database_names && i.tpe == IndexType::Unique),
+    let other_is_unique = || match &relation_field.database_names.len() {
+        0 => table.is_column_unique(&relation_field.name),
+        1 => {
+            let column_name = relation_field.database_names.first().unwrap();
+            table.is_column_unique(column_name)
         }
+        _ => table
+            .indices
+            .iter()
+            .any(|i| i.columns == relation_field.database_names && i.tpe == IndexType::Unique),
     };
+
     let arity = match relation_field.arity {
         FieldArity::Required | FieldArity::Optional if other_is_unique() => FieldArity::Optional,
         FieldArity::Required | FieldArity::Optional => FieldArity::List,
         FieldArity::List => FieldArity::Optional,
     };
 
-    let name = match arity {
-        FieldArity::List => prisma_inflector::default().pluralize(&model.name).camel_case(), // pluralize
-        FieldArity::Optional => model.name.clone().camel_case(),
-        FieldArity::Required => model.name.clone().camel_case(),
+    //if the backrelation name would be duplicate, probably due to being a selfrelation
+    let name = if model.name == other_model.name && relation_field.name == model.name.camel_case() {
+        format!("other_{}", model.name.clone().camel_case())
+    } else {
+        model.name.clone().camel_case()
```

Comment:
and previously it wasn't clashing because one of the two was pluralized, right?

Reply:
Yes